docs/decisions/0002-events-naming-and-versioning.rst

```diff
@@ -0,0 +1,91 @@
+2. Open edX events naming and versioning
+========================================
+
+Status
+------
+
+Accepted
+
+
+Context
+-------
+
+Event type hooks are an important public promise. They are a strong foundation
+of a healthy ecosystem of extensions for the Open edX world and have been
+recognized as such by the arch team of the Open edX core and also by the community.
+
+This ADR has the purpose of defining the rules to be followed when naming an
+Open edX Event with the intent of covering the use cases of:
+
+* Open edX core developers wanting to add new events.
+* Open edX core developers wanting to deprecate and eventually remove events.
+* Open edX extension developers wanting to create and maintain stable
+  applications, even when the events framework evolves and changes over time.
+* Service developers wanting to listen for events using message queues.
```

Comment:
Request:  I don' think anything you're saying here prevents that but it may be worth calling out that we're not going to solve fully for the cross service messaging case at this point.

Context:  The rest of this decision is referencing django signals but when we add cross-service messaging capabilities we'll be using a different technology to actually transmit the data.  At that point we'll probably want to have some way of transforming the event objects from something optimized for django signals to something that the cross service technology expects.

Reply:
Got it, we can make a note of this